mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt

```diff
@@ -173,24 +154,32 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
 
         override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
             super.onAuthenticationError(errorCode, errString)
-            finish()
+            finishActivity()
         }
 
         override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
             super.onAuthenticationSucceeded(result)
+            lastAuthenticationTimestamp = SystemClock.elapsedRealtime()
             contentView.isInvisible = false
             authPrompt = null
         }
 
         override fun onAuthenticationFailed() {
             super.onAuthenticationFailed()
-            finish()
+            finishActivity()
+        }
+
+        private fun finishActivity() {
```

Comment:
You can make this `protected` and remove it [here](https://github.com/openhab/openhab-android/blob/master/mobile/src/main/java/org/openhab/habdroid/background/NfcReceiveActivity.kt#L49).